Model/ModelManager.php

```diff
@@ -376,7 +386,7 @@ public function getNormalizedIdentifier($entity)
         }
 
         // the entities is not managed
-        if (!$entity /*|| !$this->getEntityManager($entity)->getUnitOfWork()->isInIdentityMap($entity) // commented for perfomance concern */) {
+        if (!$entity || !$this->getEntityManager($entity)->getUnitOfWork()->isInIdentityMap($entity)) {
```

Comment:
There's just one EntityManager (as in one database connection). We have a custom getObject() method in FooAdmin, maybe that's the cause? If not them I'm not sure what could be happening, since we're using standard Sonata methods for most of the rest of the call.

    public function getObject($id)
    {
        /** @var FooRepository $fooRepo */
        $fooRepo = $this->em->getRepository(FooRepository::CLASS_NAME);

        /** @var Foo $object */
        $object = $fooRepo->doComplicatedRetrieval($some, $options);

        $object->doSomething();
        return $object;
    }

And the entity manager in the Admin class is set via admin.yml:

    sonata.admin.foo:
        # ...
        calls:
            - [ setEntityManager, [ '@doctrine' ]]